src/Values.cpp

```diff
@@ -760,3 +760,17 @@ std::size_t vc4c::hash<vc4c::Value>::operator()(vc4c::Value const& val) const no
 {
 	return std::hash<std::string>::operator()(val.to_string());
 }
+
+Optional<SmallImmediate> Literal::converToSmallImmediate() const
+{
+  if (type == LiteralType::INTEGER)
+  {
+    uint32_t i = toImmediate();
+    if (i >= -15 && i <= 16) {
```

Comment:
These two lines are wrong. `i` is unsigned, so it cannot be negative!
Also, only the integer 0 to 15 match 1:1 to small immediate values.